File: polyglot/overwrite_from_context/py/src/runtime_template_resolver/sdk.py
from typing import Optional

from .logger import Logger
from .options import ResolverOptions
from .compute_registry import ComputeRegistry
from .context_resolver import ContextResolver

def create_registry(logger: Optional[Logger] = None) -> ComputeRegistry:
    """Factory function to create a new ComputeRegistry"""
    return ComputeRegistry(logger=logger)

def create_resolver(
    registry: Optional[ComputeRegistry] = None,
    options: Optional[ResolverOptions] = None,
    logger: Optional[Logger] = None
) -> ContextResolver:
    """Factory function to create a new ContextResolver"""
    reg = registry or create_registry(logger=logger)
    
    # If logger provided but not in options, create default options with logger
    if logger and not options:
        options = ResolverOptions(logger=logger)
    elif logger and options and not options.logger:
        # Override logger in options if passed explicitly but missing in options
        options.logger = logger
        
    return ContextResolver(registry=reg, options=options)

File: polyglot/overwrite_from_context/mjs/src/sdk.ts
import { Logger } from './logger.js';
import { ResolverOptions } from './options.js';
import { ComputeRegistry } from './compute-registry.js';
import { ContextResolver } from './context-resolver.js';

export function createRegistry(logger?: Logger): ComputeRegistry {
    return new ComputeRegistry(logger);
}

export function createResolver(
    registry?: ComputeRegistry,
    options?: ResolverOptions,
    logger?: Logger
): ContextResolver {
    const reg = registry || createRegistry(logger);

    if (logger && !options) {
        options = { logger };
    } else if (logger && options && !options.logger) {
        options.logger = logger;
    }

    return new ContextResolver(reg, options);
}
